File: VB/ConditionalAppearanceExample.Module/Controllers/ConditionalAppearanceController.vb
Imports System
Imports System.ComponentModel
Imports System.Collections.Generic
Imports System.Diagnostics
Imports System.Text

Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Actions
Imports DevExpress.Persistent.Base

Imports DevExpress.ExpressApp.ConditionalAppearance
Imports ConditionalAppearanceExample.Module.Business_Objects
Imports DevExpress.ExpressApp.Editors

Namespace ConditionalAppearanceExample.Module.Controllers
   Partial Public Class ConditionalAppearanceController
       Inherits ViewController

      Public Sub New()
         InitializeComponent()
         RegisterActions(components)
         TargetObjectType = GetType(Product)
      End Sub
      Private appearanceController As AppearanceController
      Protected Overrides Sub OnActivated()
         MyBase.OnActivated()
         appearanceController = Frame.GetController(Of AppearanceController)()
         If appearanceController IsNot Nothing Then
            AddHandler appearanceController.CustomApplyAppearance, AddressOf appearanceController_CustomApplyAppearance
            AddHandler appearanceController.AppearanceApplied, AddressOf appearanceController_AppearanceApplied
         End If
      End Sub

      Private Sub appearanceController_AppearanceApplied(ByVal sender As Object, ByVal e As ApplyAppearanceEventArgs)
         If (TypeOf View Is ListView) AndAlso (e.ItemType = AppearanceItemType.ViewItem.ToString()) AndAlso (e.ItemName = "Category") AndAlso (e.ContextObjects.Length > 0) Then
            If View.SelectedObjects.Contains(e.ContextObjects(0)) Then
               Dim formattedItem As IAppearanceFormat = TryCast(e.Item, IAppearanceFormat)
               If formattedItem IsNot Nothing Then
                  'Reset the font color of the Category property for selected objects
                  formattedItem.ResetFontColor()
               End If
            End If
         End If
      End Sub

      Protected Overridable Sub CustomizeDisabledEditorsAppearance(ByVal e As ApplyAppearanceEventArgs)
      End Sub

      Private Sub appearanceController_CustomApplyAppearance(ByVal sender As Object, ByVal e As ApplyAppearanceEventArgs)

         If e.AppearanceObject.Enabled = False Then
            'Customize the appearance of the editors disabled by a rule 
            CustomizeDisabledEditorsAppearance(e)
            'Cancel other possible conditional appearance changes,
            'applying only the rule that disables the item
            'e.Handled = true;
         End If
      End Sub
      Protected Overrides Sub OnDeactivated()
         If appearanceController IsNot Nothing Then
            RemoveHandler appearanceController.CustomApplyAppearance, AddressOf appearanceController_CustomApplyAppearance
            RemoveHandler appearanceController.AppearanceApplied, AddressOf appearanceController_AppearanceApplied
         End If
         MyBase.OnDeactivated()
      End Sub
   End Class
End Namespace

File: CS/ConditionalAppearanceExample.Module/Controllers/ConditionalAppearanceController.cs
using System;
using System.ComponentModel;
using System.Collections.Generic;
using System.Diagnostics;
using System.Text;

using DevExpress.ExpressApp;
using DevExpress.ExpressApp.Actions;
using DevExpress.Persistent.Base;

using DevExpress.ExpressApp.ConditionalAppearance;
using ConditionalAppearanceExample.Module.Business_Objects;
using DevExpress.ExpressApp.Editors;

namespace ConditionalAppearanceExample.Module.Controllers {
   public partial class ConditionalAppearanceController : ViewController {
      public ConditionalAppearanceController() {
         InitializeComponent();
         RegisterActions(components);
         TargetObjectType = typeof(Product);
      }
      private AppearanceController appearanceController;
      protected override void OnActivated() {
         base.OnActivated();
         appearanceController = Frame.GetController<AppearanceController>();
         if (appearanceController != null) {
            appearanceController.CustomApplyAppearance += new EventHandler<ApplyAppearanceEventArgs>(appearanceController_CustomApplyAppearance);
            appearanceController.AppearanceApplied += new EventHandler<ApplyAppearanceEventArgs>(appearanceController_AppearanceApplied);
         }
      }

      void appearanceController_AppearanceApplied(object sender, ApplyAppearanceEventArgs e) {
         if ((View is ListView) && (e.ItemType == AppearanceItemType.ViewItem.ToString()) && (e.ItemName == "Category") && (e.ContextObjects.Length > 0)) {
            if (View.SelectedObjects.Contains(e.ContextObjects[0])) {
               IAppearanceFormat formattedItem = e.Item as IAppearanceFormat;
               if (formattedItem != null) {
                  //Reset the font color of the Category property for selected objects
                  formattedItem.ResetFontColor();
               }
            }
         }
      }

      protected virtual void CustomizeDisabledEditorsAppearance(ApplyAppearanceEventArgs e) {      
      }

      void appearanceController_CustomApplyAppearance(object sender, ApplyAppearanceEventArgs e) {
         
         if (e.AppearanceObject.Enabled == false) {
            //Customize the appearance of the editors disabled by a rule 
            CustomizeDisabledEditorsAppearance(e);
            //Cancel other possible conditional appearance changes,
            //applying only the rule that disables the item
            e.Handled = true;
         }
      }
      protected override void OnDeactivated() {
         if (appearanceController != null) {
            appearanceController.CustomApplyAppearance -= new EventHandler<ApplyAppearanceEventArgs>(appearanceController_CustomApplyAppearance);
            appearanceController.AppearanceApplied -= new EventHandler<ApplyAppearanceEventArgs>(appearanceController_AppearanceApplied);
         }
         base.OnDeactivated();
      }
   }
}
